chef/cookbooks/monasca/recipes/common.rb

```diff
@@ -0,0 +1,118 @@
+#
+# Copyright 2017 Fujitsu LIMITED
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitation.
+
+platform = node[:monasca][:platform]
+
+monasca_pkgs = platform[:packages]
+monasca_project = platform[:keystone][:tenant]
+monasca_roles = platform[:keystone][:roles]
+
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+register_auth_hash = {
+  user: keystone_settings["admin_user"],
+  password: keystone_settings["admin_password"],
+  tenant: keystone_settings["admin_tenant"]
+}
+
+log "monasca:common" do
+  message "monasca:common packagess are #{monasca_pkgs}"
+  level :debug
+end
+log "monasca:common" do
+  message "monasca:common project is #{monasca_project}"
+  level :debug
+end
+log "monasca:common" do
+  message "monasca:common roles are #{monasca_roles}"
+  level :debug
+end
+
+monasca_pkgs.each do |pkg|
+  package pkg
+end
+
+keystone_register "monasca:common wakeup keystone" do
+  protocol keystone_settings["protocol"]
+  insecure keystone_settings["insecure"]
+  host keystone_settings["internal_url_host"]
+  port keystone_settings["admin_port"]
+  auth register_auth_hash
+  action :wakeup
+end
+
+keystone_register "monasca:common create tenant #{monasca_project} for monasca" do
+  protocol keystone_settings["protocol"]
+  insecure keystone_settings["insecure"]
+  host keystone_settings["internal_url_host"]
+  port keystone_settings["admin_port"]
+  auth register_auth_hash
+  tenant_name monasca_project
+  action :add_tenant
+end
+
+monasca_roles.each do |role|
+  keystone_register "monasca:common register #{role} role in #{monasca_project} tenant" do
+    protocol keystone_settings["protocol"]
+    insecure keystone_settings["insecure"]
+    host keystone_settings["internal_url_host"]
+    port keystone_settings["admin_port"]
+    auth register_auth_hash
+    role_name role
+    action :add_role
+  end
+end
+
+### recipes agents specific settings
+### single credentials all metrics@logs agents
+agents_settings = []
+
+if node["roles"].include?("monasca-agent")
+  agents_settings.push(node[:monasca][:agent][:keystone])
+end
+if node["roles"].include?("monasca-log-agent")
+  agents_settings.push(node[:monasca][:log_agent][:keystone])
+end
+
+unless agents_settings.empty?
+  agents_settings.each do |hulk|
```

Comment:
Big green guy whose power is so great that he made himself to crowbar-openstack :anger: